docs/installationUbuntu20.04.md

```diff
@@ -0,0 +1,167 @@
+
+```
+eval "$(ssh-agent -s)"
+```
+
+* Make sure you get "Agent pid [SOME_NUMBER]"
+```
+ssh-add -K ~/.ssh/id_rsa
+nano ~/.ssh/id_rsa.pub
+```
+* Copy this key to github (In github: User icon->Settings->SSH and GPG keys->New SSH key)
+
+## Install zsh and oh-my-zsh
+
+*Optional*
+
+```
+git clone https://github.com/robbyrussell/oh-my-zsh.git ~/.oh-my-zsh
+cp ~/.oh-my-zsh/templates/zshrc.zsh-template ~/.zshrc
+cd ~
+nano .zshrc
+```
+* Change ZSH_THEME to "agnoster"
+Save and quit nano (ctrl-o, enter, ctrl-x)
+```
+chsh -s /bin/zsh
+cd .oh-my-zsh
+sh tools/upgrade.sh
+git clone https://github.com/zsh-users/zsh-syntax-highlighting.git "$HOME/.zsh-syntax-highlighting" --depth 1
+echo "source $HOME/.zsh-syntax-highlighting/zsh-syntax-highlighting.zsh" >> "$HOME/.zshrc"
+````
+
+## Clone repos
+```
+cd ~/sources
+git clone ssh://git@github.com/Project-OSRM/osrm-backend
+git clone ssh://git@github.com/chairemobilite/trRouting
+git clone ssh://git@github.com/chairemobilite/transition
+```
+
+## Setup Postgresql
+
+```
+sudo -u postgres psql
+```
+* Choose a password: `\password`
+* Quit psql: `\q`
+
+## Install OSRM
+
+* Build and install
+```
+cd ~/sources/osrm-backend
+git checkout tags/v5.26.0
+git switch -c 5.26.0
+mkdir -p build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Release
+cmake --build .
+sudo cmake --build . --target install
+```
+
+## Install trRouting
+```
+cd ~/sources/trRouting
+autoreconf -i
+./configure
+make
+sudo make install
+````
+
+## Install and setup node and yarn
+```
+curl -sL https://deb.nodesource.com/setup_13.x | sudo bash -
+sudo apt-get install -y nodejs
+curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+sudo apt-get update && sudo apt-get install yarn
+```
+
+## Install Transition
+```
+cd ~/sources/transition
+yarn install
+cp .env.example .env
+```
+### Configure the .env file
+* Change `PG_CONNECTION_STRING_PREFIX=postgres://postgres:@localhost:5432/` to `PG_CONNECTION_STRING_PREFIX=postgres://postgres:YOUR_POSTGRES_PASSWORD@localhost:5432/`
+* Change `EXPRESS_SESSION_SECRET_KEY` to a random string with no space
+* Change `PROJECT_CONFIG` to point to your project's configuration file. The default is an example configuration file that can be copied and configured for your own need.
+
+### Get a Mapbox access token
+* Go to [Mapbox](http://mapxbox.com) and sign up
+* Go to your account dashboard, then generate a new access token
+* Open the `.env` file
+* Copy this access token to `.env` file: `MAPBOX_ACCESS_TOKEN=YOUR_TOKEN`
+* If you have a custom mapbox style, put your username and style id in `MAPBOX_USER_ID` and `MAPBOX_STYLE_ID`
+
+
+### Setup project
+```
+yarn setup
+yarn migrate
+yarn compile
+yarn create-user
+```
+If it fails and asks to locate postgis scripts file, do the following:
+
+`sudo apt-get install postgresql-12-postgis-3 postgresql-12-postgis-3-scripts`
```

Comment:
postgres version is wrong here I think. 
Also why is it needed? 
It's ont a specific version of ubuntu, it should be needed or not, not something in between

Reply:
It's what I have on the latest ubuntu 20.04... And I don't know why it's under an `if`. From my experience of one time installation on a new install, it fails, so they are needed... I guess we should install them directly at the top of the instructions.